### PR TITLE
[cirque] Add cirque test for Python device controller

### LINKED ISSUE #4645: Add e2e cirque test between python device controller and lighting example regarding thread network provisioning
#### Problem
Add e2e cirque test between python device controller and lighting example regarding thread network provisioning

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem

This is a resubmit of #5317, with #5683 included

We are missing some tests for Python binding of chip device controller, this test adds them.

Will mark as ready once cirque is re-enabled.

#### Summary of Changes

* Add cirque test for python device controller.
* Update log outputs for tests
* Wrap tests using gdb to catch potential crashes.

Fixes #4645

### CODE REVIEW

File: src/controller/python/build-chip-wheel.py [hunk 1]
```diff
@@ -135,6 +135,7 @@ def finalize_options(self):
             'chip.logging',
             'chip.native',
             'chip.tlv',
+            'chip.clusters',
```
Comment (Damian-Nordic) [inline src/controller/python/build-chip-wheel.py:138]: it's already on the list

File: src/controller/python/test/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+FROM ubuntu:focal
+
+ENV DEBIAN_FRONTEND noninteractive
+
+COPY /whl/python/*.whl /whl/
+COPY entrypoint.sh /entrypoint.sh
+COPY mobile-device-test.py /usr/bin/
+
+RUN apt-get update && \
+    apt-get install -y ca-certificates libglib2.0-dev avahi-daemon libavahi-client3 \
```
Comment (Damian-Nordic) [inline src/controller/python/test/Dockerfile:11]: Consider using  `install --no-install-recommends` to only install necessary packages.

**Review by Damian-Nordic — APPROVED** [review]

**Review by yunhanw-google — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

File: examples/lighting-app/linux/LightingManager.cpp [hunk 3]
```diff
@@ -49,13 +49,13 @@ bool LightingManager::InitiateAction(Action_t aAction)
     switch (aAction)
     {
     case ON_ACTION:
-        printf("LightingManager::InitiateAction(ON_ACTION)");
+        std::cout << "LightingManager::InitiateAction(ON_ACTION)" << std::endl;
```
Comment (mspang) [inline examples/lighting-app/linux/LightingManager.cpp:52]: Why aren't we using the logging APIs here?

**Review by mspang — CHANGES_REQUESTED** [review]
Please don't copy artifacts (.whl files) to the source tree. You can copy source files into the output directory instead.

This process looks highly nondeterministic:
- copying files with wildcards
- recursive copies
- building dockers without cleaning up stale files
- load bearing "sleep" calls all over the place

**Comment by erjiaqing** [comment]
> Please don't copy artifacts (.whl files) to the source tree. You can copy source files into the output directory instead.
> 
> This process looks highly nondeterministic:
> 
> * copying files with wildcards

- Resolved

> * recursive copies

- Resolved

> * building dockers without cleaning up stale files

- This image is local use only

> * load bearing "sleep" calls all over the place

- Removed since most of related feature is implemented, with a note about `EnableNetwork`

File: src/controller/python/test/mobile-device-test.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+
+from chip import ChipDeviceCtrl
+from chip import exceptions
+from optparse import OptionParser, OptionValueError
```
Comment (mspang) [inline src/controller/python/test/mobile-device-test.py:5]: nit: This is deprecated, please use `argparse`

File: src/test_driver/linux-cirque/helper/CHIPTestBase.py [hunk 5]
```diff
@@ -198,6 +198,17 @@ def get_device_thread_ip(self, device_id):
     def get_device_log(self, device_id):
         return self.query_api('device_log', [self.home_id, device_id], binary=True)
 
+    def wait_for_device_output(self, device_id, pattern, timeout = 1):
```
Comment (mspang) [inline src/test_driver/linux-cirque/helper/CHIPTestBase.py:201]: Can we stream the log output so we don't have to re-read it on a 1s delay ?

**Review by mspang — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]
This is great. Looking forward to having it as a part of our CI asap! 👍

**Pull request merged**